src/feature.rs

```diff
@@ -182,7 +159,26 @@ impl Serialize for Feature {
     where
         S: Serializer,
     {
-        JsonObject::from(self).serialize(serializer)
+        let mut map = serializer.serialize_map(None)?;
```

Comment:
I wanted to suggest passing in the length here, but `serde_json` doesn't seem to use it.